### ISSUE TITLE
Add Inject value and runtime custom map

### ISSUE THREAD

**Issue by spyhunter88** [expressmapper/ExpressMapper#36]
While I want to update a ViewModel to an Model/Entity and save back to Database.
1. I must keep the Entity's reference and update new value into it without create new Entity. So with current ExpressMapper function, I don't see I can update and exist object with new value from another object. Just only return new object.
2. Some of its properties can not be update base on some others value. Like with `ModelStatus = 10 && CurrentUser = "spyhunter88"`, the field CustomerName will be updated and saved to database, otherwise is not.
Like the sample in homepage, I don't think I can check like that because Context some time don't have enough information like CurrentUser = "spyhunter88".
It will be dynamic if you have some mapper like:
```
public class MyCustomMapper : ICustomMapper<ViewModel, Model>
{
  private MyConfig config;
  public MyCustomMapper(MyConfig config) { this.config = config; }

  public Model Map
        (IMappingContext<ViewModel,Model> context)
    {
        // base on config variable, I can restrict what properties will be mapped or not
        context.Destination.CustomerName = context.Source.CustName;
        return context.Destination;
    }
}
```
And use like that:
```
MyCustomMapper mapper = new MyCustomMapper(config);
Mapper.Map<ViewModel, Model>(TSrc src, TDest dest, ICustomMapper mapper);
```

This'll make my work so easy than before and may have your performance.

Also with my current project, this has some others function I have hand-code. I'll post in another issue.
This feature come with CustomMap for ICollection, what behavior when Source have new object, or missing object compare with Destination list.

**Comment by anisimovyuriy** [comment]
Dear spyhunter88,

Thanks a lot for your valuable input!
1) Actually you can do that. Use the following signature of the "Map" method: 
```Mapper.Map<ViewModel, Model>(viewModel, model);```
In that case you update your existing model object with new values from viewModel
2) Very interesting point. I think with current implementation you have 2 choices:
```
Mapper.Register<ViewModel, Model>()
.Ignore(dest => dest.CustomerName)
.After((src, dest) => {
       if (src.ModelStatus = 10 && src.CurrentUser = "spyhunter88")
       {
                 dest.CustomerName = src.CustomerName;
       }
});
```
Or you can do it via custom mapper as you provided the sample above "MyCustomMapper" but using ```Mapper.Map<ViewModel, Model>(viewModel, model);```

Please let me know if I answered to you questions correctly.

Thanks a lot spyhunter88!

**Comment by spyhunter88** [comment]
Thanks, I'll give a try from your suggestion. But in the case 2, may be I'm not clearly about my case. The condition of CurrentUser and ModelStatus is not hard-coded Integer or String. The number 10 represents for a random CurrentStatus that I pick from Database. Also "spyhunter88" is the currently user log in to system. And these are per request (I use web project).
In your first solution. I must use a condition like: `if (src.ModelStatus == currentStatus && src.CurrentUser == currentLoginUser) ....`
And I must register in each logic map, after retrieve currentStatus from Database. That's not a good idea about Register something inside logic code.

In the second solution. I must pass an instance of MyCustomMapper into Map function instead of class definition. I re-write my sample:
```
public class MyCustomMapper : ICustomMapper<ViewModel, Model>
{
  private MyConfig config;
  public MyCustomMapper(MyConfig config) { this.config = config; }

  public Model Map
        (IMappingContext<ViewModel,Model> context)
    {
        if (context.Src.ModelStatus == config.currentStatus && .... ) // '....' mean any other condition
        {
             context.Destination.CustomerName = context.Source.CustName;
        }
        else
        {
             // any others mapping
        }
        return context.Destination;
    }
}
```
So if use `Mapper.Map<ViewModel, Model>(viewModel, model);`, there's currently no way to pass MyCustomMapper instance, and no way to pass myConfig object into it, and of course, your Mapping maybe can not run because config variable inside MyCustomMapper is null, is that right?
I'll try with an static object, but that's not my prefer. Because I will not map one Model each time, I map a series of model with seperate config for each Model. My logic code I like will be:
```
MyConfig config = new MyConfig();
foreach (ViewModel vm in vms)
{
     // update config base in vm
     Utils.UpdateConfig(vm, out config);

     // passing into MyCustomMapper
     MyCustomMapper customMapper = new MyCustomMapper(config);

     // find relate Model
     Model model = Utils.FindModelByKey(vm);
     
     // run mapper finally
     Mapper.Map<ViewModel, Model>(vm, model, customMapper);

     // save it back to database
     Utils.SaveModel(model);
}
```

That's my goal for case 2.

**Comment by anisimovyuriy** [comment]
Dear spyhunter,

Regarding putting business logic into Register - completely support the comment - that was done for illustration purposes and of course it should be baked into BL class.
Regarding the second case I got you that you would like to control ICustomMapper creation and supply it during the actual mapping. I think you made a good point for improvement. Thanks! I'll add it as improvement.

**Comment by anisimovyuriy** [comment]
FIXED and will be included in 1.1 release
